src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -327,37 +328,51 @@ private static async Task<Stream> GetStream(string input, ILogger logger)
             return requestUrls;
         }
 
-        internal static async Task ValidateOpenApiDocument(string openapi, LogLevel loglevel)
+        internal static async Task ValidateOpenApiDocument(string openapi, LogLevel loglevel, CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(openapi))
-            {
-                throw new ArgumentNullException(nameof(openapi));
-            }
             var logger = ConfigureLoggerInstance(loglevel);
-            var stream = await GetStream(openapi, logger);
 
-            OpenApiDocument document;
-            logger.LogTrace("Parsing the OpenApi file");
-            document = new OpenApiStreamReader(new OpenApiReaderSettings
+            try
             {
-                RuleSet = ValidationRuleSet.GetDefaultRuleSet()
-            }
-            ).Read(stream, out var context);
+                if (string.IsNullOrEmpty(openapi))
+                {
+                    throw new ArgumentNullException(nameof(openapi));
+                }
+                var stream = await GetStream(openapi, logger, cancellationToken);
 
-            if (context.Errors.Count != 0)
-            {
-                foreach (var error in context.Errors)
+                OpenApiDocument document;
+                logger.LogTrace("Parsing the OpenApi file");
+                document = new OpenApiStreamReader(new OpenApiReaderSettings
                 {
-                    Console.WriteLine(error.ToString());
+                    RuleSet = ValidationRuleSet.GetDefaultRuleSet()
+                }
+                ).Read(stream, out var context);
+
+                if (context.Errors.Count != 0)
+                {
+                    foreach (var error in context.Errors)
+                    {
+                        logger.LogError("OpenApi Parsing error: {message}", error.ToString());
+                        Console.WriteLine(error.ToString());
```

Comment:
```suggestion
```

You're already using the logger, we should not have anymore console.log in the code base